pkg/controllers/cloud/cloudprovideraccount_controller.go

```diff
@@ -214,3 +240,87 @@ func (r *CloudProviderAccountReconciler) getAccountProviderType(namespacedName *
 
 	return r.accountProviderType[*namespacedName]
 }
+
+// getCPABySecret returns nil only when the Secret is not used by any CloudProvideAccount CR,
+// otherwise the dependent CloudProvideAccount CR will be returned.
+func (r *CloudProviderAccountReconciler) getCPABySecret(s types.NamespacedName) (error, []cloudv1alpha1.CloudProviderAccount) {
+	cpaList := &cloudv1alpha1.CloudProviderAccountList{}
+	err := r.Client.List(context.TODO(), cpaList, &client.ListOptions{})
+	var cpaItems []cloudv1alpha1.CloudProviderAccount
+	if err != nil {
+		return fmt.Errorf("failed to get CloudProviderAccount list, err:%v", err), nil
+	}
+
+	for _, cpa := range cpaList.Items {
+		if cpa.Spec.AWSConfig != nil {
+			if cpa.Spec.AWSConfig.SecretRef.Name == s.Name &&
+				cpa.Spec.AWSConfig.SecretRef.Namespace == s.Namespace {
+				cpaItems = append(cpaItems, cpa)
+			}
+		}
+
+		if cpa.Spec.AzureConfig != nil {
+			if cpa.Spec.AzureConfig.SecretRef.Name == s.Name &&
+				cpa.Spec.AzureConfig.SecretRef.Namespace == s.Namespace {
+				cpaItems = append(cpaItems, cpa)
+			}
+		}
+	}
+	return nil, cpaItems
+}
+
+// watchSecret watch the Secret objects.
+func (r *CloudProviderAccountReconciler) watchSecret() error {
+	for {
+		event, ok := <-r.watcher.ResultChan()
+		if !ok {
+			r.resetSecretWatcher()
+		} else {
+			switch event.Type {
+			case watch.Modified:
+				secret := event.Object.(*v1.Secret)
+				namespacedName := types.NamespacedName{Namespace: secret.Namespace, Name: secret.Name}
+				err, cpaItems := r.getCPABySecret(namespacedName)
+				if err != nil {
+					r.Log.Error(err, "error while getting CPA by Secret %v", secret.Name)
+					return err
+				}
+				for _, cpa := range cpaItems {
+					// If the credentials are updated to new cloud account then vpc cache is not updated. As a
+					// result vpc cache will contain vpcs from old and new accounts.
+					err := r.processCreate(&namespacedName, &cpa)
+					if err != nil {
+						r.Log.Error(err, "error while adding the secret in CPAs")
```

Comment:
Any error returned in watchSecret() will exit the controller and restart the pod.
IMO, we should not be restarting for error in L293, since only one CPA credentials update failed. May be we can just log an error and let user check the logs. 

@reachjainrahul Any suggestions here?

Reply:
```suggestion
						r.Log.Error(err, "failed to update account" , "account", accountNamespacedName)
```